src/core/annotations/qgsannotationlayer.cpp

```diff
@@ -26,6 +26,7 @@ QgsAnnotationLayer::QgsAnnotationLayer( const QString &name, const LayerOptions
   : QgsMapLayer( QgsMapLayerType::AnnotationLayer, name )
   , mTransformContext( options.transformContext )
 {
+  mShouldValidateCrs = false;
```

Comment:
@nyalldawson you are not using this anymore do you? Only checking if a layer is an annotationlayer in further commits.
So is this obsolete? Or are you planning to use it somewhere else?